ddp-workspace/projects/ddp-dsm-ui/src/app/shipping/shipping.component.ts

```diff
@@ -271,7 +274,9 @@ export class ShippingComponent implements OnInit {
     if (localStorage.getItem(ComponentService.MENU_SELECTED_REALM) != null &&
       localStorage.getItem(ComponentService.MENU_SELECTED_REALM) !== ''
     ) {
+      this.disableCheckboxes = true;
       this.dsmService.getKitRequests(localStorage.getItem(ComponentService.MENU_SELECTED_REALM), this.shippingPage, kitType.name)
+        .pipe(finalize(() => this.disableCheckboxes = false))
```

Comment:
It would be nice to add `unsubscribe` for the subscription.

Reply:
[Don't think it needs that,](https://lukaonik.medium.com/do-we-need-to-unsubscribe-http-client-in-angular-86d781522b99) as .getKitRequests() return HttpCLient get() function call.

Reply:
Generally yes.
But there is [some nuance/alternative opinion](https://medium.com/angular-in-depth/why-you-have-to-unsubscribe-from-observable-92502d5639d0#:~:text=but%20what%20would%20be%20the%20best%20way%20to%20work%20with%20HttpClient%3F).:wink:

Reply:
I am aware of that opinion. It's still not needed here, though. :D